component/core/src/main/java/org/exoplatform/social/core/listeners/SocialUserProfileEventListenerImpl.java

```diff
@@ -18,79 +18,103 @@
-      String pPosition = (String) profile.getProperty(Profile.POSITION);     
-      //
-      boolean hasUpdated = false;
-  
-      //
-      if (!StringUtils.equals(uGender, pGender)) {
-        profile.setProperty(Profile.GENDER, uGender);
-        List<Profile.UpdateType> list = new ArrayList<Profile.UpdateType>();
-        list.add(Profile.UpdateType.CONTACT);
-        profile.setListUpdateTypes(list);
-        hasUpdated = true;
-      }
-      
-      if (!StringUtils.equals(uPosition, pPosition)) {
-        profile.setProperty(Profile.POSITION, uPosition);
-        List<Profile.UpdateType> list = new ArrayList<Profile.UpdateType>();
-        list.add(Profile.UpdateType.CONTACT);
-        profile.setListUpdateTypes(list);
-        hasUpdated = true;
-      }
-  
-      if (hasUpdated && !isNew) {
-        IdentityManager identityManager = CommonsUtils.getService(IdentityManager.class);
-        identityManager.updateProfile(profile);
-      }
-      
+    Identity identity = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, userProfile.getUserName());
+    Profile profile = identity.getProfile();
+    String uGender = userProfile.getAttribute(UserProfile.PERSONAL_INFO_KEYS[4]);// "user.gender"
+    String uPosition = userProfile.getAttribute(UserProfile.PERSONAL_INFO_KEYS[7]);// user.jobtitle
+    String pGender = (String) profile.getProperty(Profile.GENDER);
+    String pPosition = (String) profile.getProperty(Profile.POSITION);
+
+    AtomicBoolean hasUpdated = new AtomicBoolean(false);
+    Map<String, String> properties = userProfile.getUserInfoMap();
+    exlcudedAttributeList.forEach(properties.keySet()::remove);
+    properties.forEach((name, value) -> {
+      updateProfilePropertySettings(name, profilePropertySettingsService);
       if (isNew) {
-        IdentityStorage identityStorage = CommonsUtils.getService(IdentityStorage.class);
-        identityStorage.updateProfile(profile);
+        profile.setProperty(name, value);
+      } else if (!StringUtils.equals((String) profile.getProperty(name), userProfile.getAttribute(name))) {
+        profile.setProperty(name, value);
+        hasUpdated.set(true);
+      }
+    });
+
+    if (!StringUtils.equals(uGender, pGender)) {
+      profile.setProperty(Profile.GENDER, uGender);
+      hasUpdated.set(true);
+    }
+    if (!StringUtils.equals(uPosition, pPosition)) {
+      profile.setProperty(Profile.POSITION, uPosition);
+      hasUpdated.set(true);
+    }
+
+    if (hasUpdated.get()) {
+      List<Profile.UpdateType> updateTypes = new ArrayList<>();
+      updateTypes.add(Profile.UpdateType.CONTACT);
+      profile.setListUpdateTypes(updateTypes);
+    }
+
+    if (hasUpdated.get() && !isNew) {
+      identityManager.updateProfile(profile);
+    }
+
+    if (isNew) {
+      IdentityStorage identityStorage = CommonsUtils.getService(IdentityStorage.class);
+      identityStorage.updateProfile(profile);
+    }
```

Comment:
Code readability and layers usage, see comment : 

```suggestion
    if (isNew) {
      IdentityStorage identityStorage = CommonsUtils.getService(IdentityStorage.class);
      // Fixme: Avoid calling updateProfile using Service layer to avoid calling listeners again
      // Should have a dedicated method on Service layer, like 'updateProfileNoBroadcast'
      // to allow it without having to bypass it and directly interact with Storage layer
      identityStorage.updateProfile(profile);
    } else if (hasUpdated.get()) {
      identityManager.updateProfile(profile);
    }
```